From 0a5c30820a89ea9e9a07e357dc96f5d1b2bdea4f Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 21 Jun 2017 17:04:40 +0800
Subject: [PATCH 1986/5242] MLK-15135-1: dma: imx-sdma: fix build warning on
 aarch64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  d1b6e3e9c028b088f40a90acc7694c0cf540b586 from
https://source.codeaurora.org/external/imx/linux-imx.git

fix below build warning with aarch64:
drivers/dma/imx-sdma.c: In function ‘sdma_prep_dma_cyclic’:
drivers/dma/imx-sdma.c:1727:3: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘size_t’ [-Wformat=]
   dev_dbg(sdma->dev, "entry %d: count: %d dma: %pad %s%s\n",

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index d3860d4..903a197 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -9,6 +9,7 @@
 // Based on code from Freescale:
 //
 // Copyright 2004-2016 Freescale Semiconductor, Inc. All Rights Reserved.
+// Copyright 2017 NXP.
 
 #include <linux/init.h>
 #include <linux/iopoll.h>
@@ -1580,7 +1581,7 @@ static struct dma_async_tx_descriptor *sdma_prep_memcpy(
 			param &= ~BD_CONT;
 		}
 
-		dev_dbg(sdma->dev, "entry %d: count: %d dma: 0x%u %s%s\n",
+		dev_dbg(sdma->dev, "entry %d: count: %zd dma: 0x%x %s%s\n",
 				i, count, bd->buffer_addr,
 				param & BD_WRAP ? "wrap" : "",
 				param & BD_INTR ? " intr" : "");
@@ -1753,8 +1754,8 @@ static struct dma_async_tx_descriptor *sdma_prep_dma_cyclic(
 		if (i + 1 == num_periods)
 			param |= BD_WRAP;
 
-		dev_dbg(sdma->dev, "entry %d: count: %zu dma: %#llx %s%s\n",
-				i, period_len, (u64)dma_addr,
+		dev_dbg(sdma->dev, "entry %d: count: %zu dma: %pad %s%s\n",
+				i, period_len, &dma_addr,
 				param & BD_WRAP ? "wrap" : "",
 				param & BD_INTR ? " intr" : "");
 
-- 
1.7.9.5

